dvc/scm/git/__init__.py

```diff
@@ -282,8 +291,8 @@ def get_diff_trees(self, a_ref, b_ref=None):
         """
         diff_dct = {DIFF_EQUAL: False}
         trees, commit_refs = self._get_diff_trees(a_ref, b_ref)
-        diff_dct[DIFF_A_REF] = commit_refs[0]
-        diff_dct[DIFF_B_REF] = commit_refs[1]
+        diff_dct[DIFF_A_REF] = commit_refs[0][:7]
```

Comment:
Maybe we should use something like `git rev-parse --short` instead, so we are not hardcoding 7 here? `rev-parse` should increase/decrease the length as needed.

Reply:
Yes that sounds more correct. I'll try it, thanks.

Reply:
Unfortunately theres' nothing like `rev-parse --short` in gitpython that I can find. Only the current way: printing the string representation of the commit using [__to_str__()](https://gitpython.readthedocs.io/en/stable/reference.html#git.objects.base.Object.__str__) which...

Reply:
@jorgeorpinel There is:
```
import git
g = git.Repo(".")
g.git.rev_parse('HEAD', short=True)
```

Reply:
Thanks @efiop ! I did see [rev_parse()](https://gitpython.readthedocs.io/en/stable/reference.html#git.repo.base.Repo.rev_parse) in the Repo object but that doesn't accept `short`. I didn't realize you can get a more direct `git` object from it! (I don't think it's documented e.g. https://gitpython.readthedocs.io/en/stable/reference.html#git.repo.base.Repo.git)

I'll change it removing the hardcoded `[:7]` (and in all the tests too...)

Reply:
Ok, updated in 9516e70 but I have a few comments. Will open new conversations...